aicsimageio/readers/array_like_reader.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from typing import List, Optional, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from .. import constants, exceptions
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..types import MetaArrayLike
+from .reader import Reader
+
+###############################################################################
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class ArrayLikeReader(Reader):
+    """
+    A catch all for numpy, dask, or xarray to Reader interface.
+
+    Parameters
+    ----------
+    image: Union[List[MetaArrayLike], MetaArrayLike]
+        A single, numpy ndarray, dask Array, or xarray DataArray, or list of many.
+        If provided a list, each item in the list will be exposed through the scene API.
+        If provided an xarray DataArray alone or as an element of the list, the
+        known_dims and known_channel_names, kwargs are ignored if there is dimension
```

Comment:
Does this override happen if both the xarray and the known_dims are populated?

Reply:
Nope will override either independently. Will note this.